core-bundle/src/DependencyInjection/Compiler/RegisterPagesPass.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Contao\CoreBundle\Routing\Page\CompositionAwareInterface;
+use Contao\CoreBundle\Routing\Page\PageRegistry;
+use Contao\CoreBundle\Routing\Page\PageRouteEnhancerInterface;
+use Contao\CoreBundle\Routing\Page\RouteConfig;
+use Contao\FrontendIndex;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers Contao pages in the registry.
+ */
+class RegisterPagesPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private const TAG_NAME = 'contao.page';
+
+    /**
+     * Adds the fragments to the registry.
+     */
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->has(PageRegistry::class)) {
+            return;
+        }
+
+        $this->registerPages($container);
+    }
+
+    /**
+     * @throws InvalidConfigurationException
```

Comment:
The method does not throw an exception, so this should be removed IMHO.

Reply:
Removed in a972f8d81da4a457cfd3dcadcb32c77e1d3c88aa